lib/swoosh/adapters/gmail.ex

```diff
@@ -143,4 +144,12 @@ defmodule Swoosh.Adapters.Gmail do
 
   defp prepare_reply_to(body, %{reply_to: nil}), do: body
   defp prepare_reply_to(body, %{reply_to: reply_to}), do: Mail.put_reply_to(body, reply_to)
+
+  defp prepare_custom_headers(body, %{headers: nil}), do: body
```

Comment:
```suggestion
  defp prepare_custom_headers(body, %{headers: headers}) when headers == %{}, do: body
```

headers default to an empty map.